source/ucumLhcUtils.js

```diff
@@ -263,40 +350,79 @@ export class UcumLhcUtils {
 
         if (fromUnit && toUnit) {
           try {
-            // if no molecular weight was specified perform a normal conversion
-            if (!molecularWeight) {
-              returnObj['toVal'] = toUnit.convertFrom(fromVal, fromUnit);
+            const convertType = this.detectConversionType(fromUnit, toUnit); 
+
+            switch (convertType) {
```

Comment:
Looks good. If you want, you could combine similar cases when it makes sense/is worth it, e.g.,:
```
case mol->mass:
case mass->mole:
    check molecularWeight.
    check is-commensurable
    returnObj.toVal = convertType === 'mol->mass'? 
        from.toMass(): 
        from.toMol();
```